File: legacy/src/com/cattailsw/nanidroid/install/NarArchiveInventoryResult.java
package com.cattailsw.nanidroid.install;

/** Frozen Ant-build compatibility copy; the Gradle lane is implemented in Kotlin. */
public final class NarArchiveInventoryResult {
    private final NarArchiveInventory inventory;
    private final NarInstallError error;
    private final String detail;

    private NarArchiveInventoryResult(
            NarArchiveInventory inventory, NarInstallError error, String detail) {
        this.inventory = inventory;
        this.error = error;
        this.detail = detail;
    }

    static NarArchiveInventoryResult success(NarArchiveInventory inventory) {
        return new NarArchiveInventoryResult(inventory, null, "");
    }

    static NarArchiveInventoryResult failure(
            NarInstallError error, String detail) {
        return new NarArchiveInventoryResult(null, error, detail);
    }

    public boolean isSuccess() { return inventory != null; }
    public NarArchiveInventory getInventory() { return inventory; }
    public NarInstallError getError() { return error; }
    public String getDetail() { return detail; }
}

File: src/main/kotlin/com/cattailsw/nanidroid/install/NarArchiveInventoryResult.kt
package com.cattailsw.nanidroid.install

/** Immutable result of structural NAR inventory validation. */
class NarArchiveInventoryResult private constructor(
    private val inventory: NarArchiveInventory?,
    private val error: NarInstallError?,
    private val detail: String,
) {
    fun isSuccess(): Boolean = inventory != null

    fun getInventory(): NarArchiveInventory? = inventory

    fun getError(): NarInstallError? = error

    fun getDetail(): String = detail

    companion object {
        @JvmStatic
        fun success(
            inventory: NarArchiveInventory,
        ): NarArchiveInventoryResult = NarArchiveInventoryResult(
            inventory,
            null,
            "",
        )

        @JvmStatic
        fun failure(
            error: NarInstallError,
            detail: String,
        ): NarArchiveInventoryResult = NarArchiveInventoryResult(
            null,
            error,
            detail,
        )
    }
}
